modules/actions/snapshots/lib/compute-daily-values.ts

```diff
@@ -0,0 +1,31 @@
+import { Prisma } from '@prisma/client';
+import _ from 'lodash';
+
+export const computeDailyValues = (
+    snapshots: Prisma.PrismaPoolSnapshotUncheckedCreateInput[],
+): Prisma.PrismaPoolSnapshotUncheckedCreateInput[] => {
+    // Group snapshots by poolId
+    const groupedByPoolId = _.groupBy(snapshots, 'poolId');
+
+    // Iterate through each group to compute daily values
+    const updatedSnapshots = _.flatMap(groupedByPoolId, (snapshots) => {
+        // Sort snapshots by timestamp for the pool
+        const sortedSnapshots = _.sortBy(snapshots, 'timestamp');
+
+        return sortedSnapshots.map((snapshot, index) => {
+            const previousSnapshot = sortedSnapshots[index - 1];
```

Comment:
doesnt this return the wrong one in the first iteration when index is `0`?

Reply:
than it's undefined and falls back to 0